Documentation/disaster-recovery.md

```diff
@@ -106,3 +106,102 @@ kubectl create -f operator.yaml
+10. Run `kubectl exec -it rook-ceph-mon-a-XXXXXX bash` to enter your `ceph-mon` pod,
+
+```shell
+mon-a# cat /etc/ceph/keyring-store/keyring  # save this keyring content, for later use
+mon-a# exit
+```
+
+11. Run `kubectl edit deploy/rook-ceph-operator` and set `replicas` to 0.
+12. Run `kubectl delete deploy/X` where X is every deployment in namespace `rook-ceph`, except `rook-ceph-operator` and `rook-ceph-tools`.
+
+SSH to the host where `rook-ceph-mon-a` in your new Kubernetes cluster resides.
+
+13. Pick the latest `ceph-mon` directory (`/var/lib/rook/mon-?`) in your previous backup, replace `/var/lib/rook/mon-a` with it.
+14. Replace `/var/lib/rook/mon-a/keyring` with the saved keyring, preserving only the `[mon.]` section, remove `[client.admin]` section.
+15. Get your `rook-ceph-mon-a` address by `kubectl get cm/rook-ceph-mon-endpoints -o yaml` in your new Kubernetes cluster.
+16. Run `docker run -it --rm -v /var/lib/rook:/var/lib/rook ceph/ceph:v14.2.1-20190430 bash` (note the docker images version, should match your deployment):
+
+```shell
+container# cd /var/lib/rook
+container# ceph-mon --extract-monmap m --mon-data ./mon-a/data
+container# monmaptool --print m
+container# monmaptool --rm a m  # repeat this until all the old ceph-mons are removed
+container# monmaptool --add a 10.77.2.216:6789 m   # Replace with your own rook-ceph-mon address!
+container# ceph-mon --inject-monmap m --mon-data ./mon-a/data
+container# rm m
+container# exit
+```
+
+Now back to your local machine.
+
+17. Run `kubectl edit secret/rook-ceph-mon` and modify `fsid` to your original `fsid`
+18. Run `kubectl edit cm/rook-config-override` add content below:
+
+```
+data:
+  config: |
+    [global]
+    auth cluster required = none
+    auth service required = none
+    auth client required = none
+    auth supported = none
+```
+
+19. Run `kubectl edit deploy/rook-ceph-operator` and set `replica` to 1.
+20. Run `kubectl logs -f rook-ceph-operator-xxxxxxxxxx` and wait till all the things are settled.
+21. **STATE:** Now your rook/ceph cluster should be up and running, with authentication disabled.
+22. Run `kubectl exec -it rook-ceph-tools-XXXXXXX bash` to enter tools pod:
+
+```shell
+tools# vi key
+[paste keyring content saved before, preserving only `[client admin]` section]
+tools# ceph auth import -i key
+tools# rm key
+```
+
+23. Run `kubectl edit cm/rook-config-override` and remove previously added configurations.
+24. Run `kubectl edit deploy/rook-ceph-operator` and set `replicas` to 0.
+25. Run `kubectl delete deploy/X` where X is every deployment in namespace `rook-ceph`, except `rook-ceph-operator` and `rook-ceph-tools`, again. This time OSD daemons are present and included.
+26. Run `kubectl edit deploy/rook-ceph-operator` and set `replicas` to 1.
+27. Run `kubectl logs -f rook-ceph-operator-xxxxxxxxxx` and wait till all the things are settled.
```

Comment:
Expand on "till all the things are settled. I think I had a suggestion on a similar line earlier in the doc.

Reply:
fixed